doc/CONTRIBUTING.md

```diff
@@ -0,0 +1,18 @@
+### Contributing
+
+Before you submit a pull request:
+
+0. In general, you should develop off of 'master' unless you are making a hotfix ('stable' is the current production-ready state).
+1. Make sure your code conforms to the [Style Guide](doc/style.md).
+2. Reference any issues that you are fixing in the commit message (ideally with
+   the [magic auto-closing syntax](https://help.github.com/articles/closing-issues-via-commit-messages)).
+3. Make sure the [tests pass](https://developer.mozilla.org/en-US/Add-ons/SDK/Tools/cfx#cfx_test).
+   Please add tests for any new functionality.
+4. BONUS: make sure your changes are compatible with the current [Firefox ESR](https://www.mozilla.org/en-US/firefox/organizations/all/), [Beta](https://www.mozilla.org/en-US/firefox/beta/), and [Aurora](https://www.mozilla.org/en-US/firefox/aurora/) releases as well as the [current stable release](https://www.mozilla.org/en-US/firefox/new/)!
+
+You can use `cfx -b` to choose different Firefox binaries. So for instance
+
+        cfx -b /path/to/firefox-aurora run
+
+can be used to test your changes on Aurora if you have its binary available
+at `/path/to/firefox-aurora`.
```

Comment:
This is very useful information, I think it should be included but in a seperate document. Several times when I've been motivated to contribute as a volunteer on an OSS project, but I was turned away by strict contribution requirements. 

I think the main document should be short, and link out to other documents rather than including them. So maybe this document could be linked to from a bullet point like:
 * we have tips for making a great pull request [here]()

Other items here could be things like "how to get useful debugging info", "how to run tests", "common issues", "how to debug a site", etc.

These are loosely laid out in #1195

I also prefer framing documentation as suggestions rather than requirements.